Client/Frontend/Browser/TabManager.swift

```diff
@@ -379,9 +381,41 @@ extension TabManager {
 }
 
 extension TabManager : WKNavigationDelegate {
+    func webView(webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
+        if tabsNetworkActivity[webView] == nil {
+            tabsNetworkActivity[webView] = false
+        }
+        if tabsNetworkActivity[webView] == false {
+            tabsNetworkActivity[webView] = true
+            incrementNetworkActivityCount()
+        }
+    }
+
     func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
+        tabsNetworkActivity[webView] = false
+        decrementNetworkActivityCount()
+
         storeChanges()
     }
+
+    func webView(webView: WKWebView, didFailNavigation navigation: WKNavigation!, withError error: NSError) {
+        tabsNetworkActivity[webView] = false
+        decrementNetworkActivityCount()
+    }
+
+    func incrementNetworkActivityCount() {
```

Comment:
I think you also need to take closing tabs into account. Can you test this with a page that takes a long time to load, close the tab while it is loading, and then see if the above delegate methods are being called? I'm not sure if they will be.